src/backend/distributed/commands/table.c

```diff
@@ -593,59 +616,213 @@ PreprocessAlterTableStmt(Node *node, const char *alterTableCommand)
 
 
 /*
- * ErrorIfAlterTableDefinesFKeyFromPostgresToCitusLocalTable errors out if
- * given ALTER TABLE statement defines foreign key from a postgres local table
- * to a citus local table.
+ * AlterTableDefinesFKeyBetweenPostgresAndNonDistTable returns true if given
+ * alter table command defines foreign key between a postgres table and a
+ * reference or citus local table.
  */
-static void
-ErrorIfAlterTableDefinesFKeyFromPostgresToCitusLocalTable(
-	AlterTableStmt *alterTableStatement)
+static bool
+AlterTableDefinesFKeyBetweenPostgresAndNonDistTable(AlterTableStmt *alterTableStatement)
 {
-	List *commandList = alterTableStatement->cmds;
+	bool rightHasPostgresLocalTable = false;
```

Comment:
Following the 4 variables (`rightHasPostgresLocalTable`, `rightHasNonDistTable`, `leftIsNonDistTable`, `leftIsPostgresTable`) and their combinations are hard to follow.

Can we avoid variable names and come-up with something similar to the below?
```C

static bool
AlterTableDefinesFKeyBetweenPostgresAndNonDistTable(AlterTableStmt *alterTableStatement)
{
	bool rightHasPostgresLocalTable = false;
	bool rightHasNonDistTable = false;

	Oid rightRelationId = InvalidOid;
	List *rightRelationIdList =
		GetAlterTableAddFKeyRightRelationIdList(alterTableStatement);

	LOCKMODE lockmode = AlterTableGetLockLevel(alterTableStatement->cmds);
	Oid leftRelationId = AlterTableLookupRelation(alterTableStatement, lockmode);
	if (IsCitusTable(leftRelationId))
	{
		return RelationIdListContains(rightRelationIdList, CITUS_TABLE_WITH_NO_DIST_KEY);
	}
	else if (IsCitusTableType(leftRelationId, CITUS_TABLE_WITH_NO_DIST_KEY))
	{
		return !RelationIdListContains(rightRelationIdList, ANY_CITUS_TABLE_TYPE);
	}

	return false;
}
```

Reply:
what about this ? (to be more explicit for postgres tables in `else if`)

```c
static bool
AlterTableDefinesFKeyBetweenPostgresAndNonDistTable(AlterTableStmt *alterTableStatement)
{
	List *rightRelationIdList =
		GetAlterTableAddFKeyRightRelationIdList(alterTableStatement);

	LOCKMODE lockmode = AlterTableGetLockLevel(alterTableStatement->cmds);
	Oid leftRelationId = AlterTableLookupRelation(alterTableStatement, lockmode);
	if (!IsCitusTable(leftRelationId))
	{
		return RelationIdListContainsCitusTableType(rightRelationIdList, CITUS_TABLE_WITH_NO_DIST_KEY);
	}

	if (IsCitusTableType(leftRelationId, CITUS_TABLE_WITH_NO_DIST_KEY))
	{
		return RelationIdListContainsPostgresTable(rightRelationIdList);
	}

	return false;
}
```

Reply:
e75d64d40d77bd98ab5574eb23ab57b376b5e4f2